src/compiler.ts

```diff
@@ -1019,13 +1022,16 @@ export class Compiler extends DiagnosticEmitter {
       let numLocals = locals.length;
       let varTypes = new Array<TypeRef>(numLocals);
       for (let i = 0; i < numLocals; ++i) varTypes[i] = locals[i].type.toRef();
-      module.addFunction(
+      const funcRef = module.addFunction(
         startFunction.internalName,
         startSignature.paramRefs,
         startSignature.resultRefs,
         varTypes,
         module.flatten(startFunctionBody)
       );
+      if (!this.options.willOptimize) {
```

Comment:
Could you sync with main and use `options.debugInfo` instead `!this.options.willOptimize` in all places?